test/integration/relations.test.js

```diff
@@ -164,4 +164,247 @@ describe.skip('Relations', function () {
+                const query = makeQuery(queryJSON);
+
+                // Check any intermediate values
+                console.log(query.toQuery());
+
+                // Perform the query against the DB
+                return query.select()
+                    .then((result) => {
+                        console.log(result);
+
+                        // NOTE: make sure to count in posts with no tags
+                        //       do not count in posts with multiple tags containing one of the excluded tags
+                        result.should.be.an.Array().with.lengthOf(3);
+
+                        // Check we get the right data
+                        // result.should.do.something;
+                    });
+            });
+
+            it('can match array $nin (single value)', function () {
+                const queryJSON = {'tags.slug': {$nin: ['animal']}};
+
+                // Use the queryJSON to build a query
+                const query = makeQuery(queryJSON);
+
+                // Check any intermediate values
+                console.log(query.toQuery());
+
+                // Perform the query against the DB
+                return query.select()
+                    .then((result) => {
+                        console.log(result);
+
+                        // NOTE: make sure to count in posts with no tags
+                        //       do not count in posts with multiple tags containing one of the excluded tags
+                        result.should.be.an.Array().with.lengthOf(3);
+
+                        // Check we get the right data
+                        // result.should.do.something;
+                    });
+            });
+
+            it('can match array $nin (multiple values)', function () {
+                const queryJSON = {'tags.slug': {$nin: ['animal', 'classic']}};
+
+                // Use the queryJSON to build a query
+                const query = makeQuery(queryJSON);
+
+                // Check any intermediate values
+                console.log('query', query.toQuery());
+
+                // Perform the query against the DB
+                return query.select()
+                    .then((result) => {
+                        console.log(result);
+
+                        // NOTE: make sure to count in posts with no tags
+                        //       do not count in posts with multiple tags containing one of the excluded tags
+                        result.should.be.an.Array().with.lengthOf(1);
+
+                        // Check we get the right data
+                        // result.should.do.something;
+                    });
+            });
+        });
+
+        // NOTE: only $lt and $gt operators are supported by nql-lang, no need to support $eq operator for now
+        describe('count', function () {
+            it('can compare by count $gt', function () {
+                const queryJSON = {'authors.posts': {$gt: 0}};
```

Comment:
IMO this must be count.posts?

Reply:
the base model to `makeQuery` atm is made with `posts`, so this becomes `count.authors` I think

Reply:
exactly, i put the the comment on the wrong use case 👍